File: x1104-Nashorn-CallingJavaMethods/src/calculator.js


var plus = function(x, y) {
    print('plus(' + x + ', ' + y + ')');
	var cls = Java.type('appl.Calculator');
	return cls.plus(x, y);
};

var minus = function(obj, x, y) {
    print('minus(' + obj + ', ' + x + ', ' + y + ')');
	return obj.minus(x, y);
};

var times = function(x, y) {
    print('times(' + x + ', ' + y + ')');
	var cls = Java.type('appl.Calculator');
	var obj = new cls();
	
	return obj.times(x, y);
};

File: x1104-Nashorn-CallingJavaMethods/src/appl/Calculator.java
package appl;


public class Calculator {

    public static int plus(int x, int y) {
        System.out.println("Calculator.plus(" + x + ", " + y + ")");
        return x + y;
    }

    public int minus(int x, int y) {
        System.out.println(this + ".minus(" + x + ", " + y + ")");
        return x - y;
    }

    public int times(int x, int y) {
        System.out.println(this + ".times(" + x + ", " + y + ")");
        return x * y;
    }
}
